nomad/structs/structs.go

```diff
@@ -54,6 +54,7 @@ type MessageType uint8
 const (
 	NodeRegisterRequestType MessageType = iota
 	NodeDeregisterRequestType
+	NodeDeregisterBatchRequestType // QUESTION does iota make it important to append this list?
```

Comment:
Yes - this needs to be appended at the end.  golang compiler would assign 0 to iota and then increments for subsequent lines.  See https://play.golang.org/p/7ZqlecuVJHW and https://golang.org/ref/spec#Iota about iota and constants.

In raft, we store the message type uint8 in messages, so these order here must remain unaltered; otherwise, we will not interpret old raft messages properly on upgrade.  Probably belongs to raft notes too.